### PR TITLE
Add links to protocol docs in SDK docs - closes #598

### LINKED ISSUE #598: Add links to protocol docs
**Describe the solution you'd like**
Now that the Lisk protocol docs have been added to the documentation, we can refer to them in the SDK docs, where adequate.

### PR DESCRIPTION
**Pull request by chris529**
Closes issue #598

### CODE REVIEW

File: modules/ROOT/pages/guides/app-development/customize.adoc [hunk 1]
```diff
@@ -152,13 +153,13 @@ TIP: For more information on creating your own custom transaction, please xref:{
...
 ====
-Transaction types 0-12 are reserved for the Lisk protocol.
+Transaction types 0-12 are reserved for the xref:{url_lisk_protocol}[Lisk Protocol].
 Do not use these to register custom transactions.
 ====
 
 Each default transaction type implements a different use case of the Lisk network.
 
-TIP: For a complete list of all default transaction types, please see the section Lisk transactions of the Lisk Protocol.
+TIP: For a complete list of all default transaction types, please see the section xref:{url_lisk_protocol_transactions}[Lisk Protocol transactions] of  the xref:{url_lisk_protocol}[Lisk Protocol].
```
Comment (Tschakki) [inline modules/ROOT/pages/guides/app-development/customize.adoc]:
```suggestion
TIP: For a complete list of all default transaction types, please see the section xref:{url_lisk_protocol_transactions}[transactions of  the Lisk protocol].
```

File: modules/ROOT/pages/guides/app-development/customize.adoc [hunk 2]
```diff
@@ -7,7 +7,8 @@ Mona Bärenfänger <mona@lightcurve.io>
 :url_github_base_transaction: https://github.com/LiskHQ/lisk-sdk/blob/{v_sdk}/elements/lisk-transactions/src/base_transaction.ts
 :url_github_sdk_entities_account: https://github.com/LiskHQ/lisk-sdk/blob/{v_sdk}/framework/src/components/storage/entities/account.js
 :url_github_sdk_entities_tx: https://github.com/LiskHQ/lisk-sdk/blob/{v_sdk}/framework/src/components/storage/entities/transaction.js
-
+:url_lisk_protocol: {v_protocol}@lisk-protocol::lisk-protocol/introduction.adoc
```
Comment (Tschakki) [inline modules/ROOT/pages/guides/app-development/customize.adoc]:
```suggestion
:v_protocol: master

:url_lisk_protocol: {v_protocol}@lisk-protocol::lisk-protocol/introduction.adoc
```

File: modules/ROOT/pages/guides/app-development/customize.adoc [hunk 3]
```diff
@@ -152,13 +153,13 @@ TIP: For more information on creating your own custom transaction, please xref:{
 
 [IMPORTANT]
 ====
-Transaction types 0-12 are reserved for the Lisk protocol.
+Transaction types 0-12 are reserved for the xref:{url_lisk_protocol}[Lisk Protocol].
```
Comment (Tschakki) [inline modules/ROOT/pages/guides/app-development/customize.adoc]:
```suggestion
Transaction types 0-12 are reserved for the xref:{url_lisk_protocol}[Lisk protocol].
```

File: modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -21,6 +21,7 @@ Mona Bärenfänger <mona@lightcurve.io>
 :url_guides: guides/index.adoc
 :url_setup: setup.adoc
 :url_tutorials: tutorials/index.adoc
+:url_lisk_protocol: {v_protocol}@lisk-protocol::lisk-protocol/introduction.adoc
```
Comment (Tschakki) [inline modules/ROOT/pages/index.adoc]:
```suggestion
:v_protocol: master

:url_lisk_protocol: {v_protocol}@lisk-protocol::lisk-protocol/introduction.adoc
```

**Review by Tschakki — CHANGES_REQUESTED** [review]
Please look for more places, where the Lisk protocol docs can be referred. And try to refer to the different pages of the protocol, where it fits int he context.

**Review by chris529 — COMMENTED** [review]
Cool...well spotted Mona! Yes..these are some good valid areas to further mention the appropriate corresponding pages in the Lisk protocol docs 👍

**Review by Tschakki — DISMISSED** [review]
LGTM

File: modules/ROOT/pages/guides/node-management/forging.adoc [hunk 5]
```diff
@@ -2,19 +2,27 @@
...
+:url_protocol_dpos: {v_protocol}@lisk-protocol::consensus-algorithm.adoc
+:url_protocol_delegate_selection: {v_protocol}@lisk-protocol::consensus-algorithm.adoc#delegate_selection
+:url_protocol_blocks_punishment: {v_protocol}@lisk-protocol::consensus-algorithm.html#punishment
+
+:url_protocol_blocks_rewards: {v_protocol}@lisk-protocol::blocks.html#rewards_and_fees
+
 == What is forging?
 
-Forging in DPoS networks is the same as mining in a PoW network: It is the process of adding new blocks to the chain.
+Forging in xref:{url_protocol_dpos}[DPoS] networks is the same as mining in a PoW network: It is the process of adding new blocks to the chain.
```
Comment (chris529) [inline modules/ROOT/pages/guides/node-management/forging.adoc]:
```suggestion
Forging in xref:{url_protocol_dpos}[DPoS] networks is the same as mining in a PoW network, which is the process of adding new blocks to the chain.
```

**Review by chris529 — COMMENTED** [review]
Looks very good, just one very minor point 👍

**Review by Tschakki — APPROVED** [review]

**Pull request merged**